dockerfiles/Dockerfile.rocm

```diff
@@ -1,24 +0,0 @@
-# --------------------------------------------------------------
-# Copyright (c) Microsoft Corporation. All rights reserved.
-# Licensed under the MIT License.
-# --------------------------------------------------------------
-# Dockerfile to run ONNXRuntime with ROCm integration
-#--------------------------------------------------------------------------
-
-FROM rocm/pytorch:rocm6.2.3_ubuntu22.04_py3.10_pytorch_release_2.3.0
-
-ARG ONNXRUNTIME_REPO=https://github.com/Microsoft/onnxruntime
-ARG ONNXRUNTIME_BRANCH=main
-
-WORKDIR /code
-
-ENV PATH=/code/cmake-3.27.3-linux-x86_64/bin:${PATH}
-
-# Prepare onnxruntime repository & build onnxruntime
-RUN git clone --single-branch --branch ${ONNXRUNTIME_BRANCH} --recursive ${ONNXRUNTIME_REPO} onnxruntime &&\
-    /bin/sh onnxruntime/dockerfiles/scripts/install_common_deps.sh &&\
-    cd onnxruntime &&\
-    /bin/sh ./build.sh --allow_running_as_root --config Release --build_wheel --update --build --parallel --cmake_extra_defines\
-            ONNXRUNTIME_VERSION=$(cat ./VERSION_NUMBER) --use_rocm --rocm_home=/opt/rocm &&\
```

Comment:
should build.py `--use_rocm` and related options also be removed?

Reply:
`--use_rocm` is used by migraphx docker as well:
https://github.com/microsoft/onnxruntime/blob/458e1bb380f211c4a2ac313c40eb680a7be66fdd/dockerfiles/Dockerfile.migraphx#L25
Keep it for now. 

@TedThemistokleous, could  you help consolidate --use_rocm and --use_migraphx, and keep only --use_migraphx?